[Bug] Failed to build with OpenSSL 4.0.0

### Search before asking

- [x] I searched in the [issues](https://github.com/apache/pulsar-client-cpp/issues) and found nothing similar.


### Version

pulsar v4.1.0
openssl 4.0.0

### Minimal reproduce step

build pulsar-client-cpp

### What did you expect to see?

build completed successfully

### What did you see instead?

In file included from /home/sdlzx/code/pulsar-client-cpp/lib/BatchMessageContainerBase.cc:22:
/home/sdlzx/code/pulsar-client-cpp/lib/MessageCrypto.h:23:10: fatal error: openssl/engine.h: No such file or directory
   23 | #include <openssl/engine.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.

### Anything else?

OpenSSL 4.0.0 has removed engine support.
reference: https://openssl-library.org/post/2026-04-14-openssl-40-final-release/

And I don't think pulsar requires this, in fact, pulsar can be built without this line.

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

--- a/lib/MessageCrypto.h
+++ b/lib/MessageCrypto.h
@@ -20,7 +20,6 @@
 #define LIB_MESSAGECRYPTO_H_
 
 #include <openssl/bio.h>
-#include <openssl/engine.h>
 #include <openssl/evp.h>
 #include <openssl/rand.h>
 #include <openssl/rsa.h>

